File: problems/p/pieceofcake2/pieceofcake2.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""
import operator
from functools import reduce

def main():
    n, h, v = [int(val) for val in input().split()]

    volume = [4]
    if h > n - h:
        volume.append(h)
    else:
        volume.append(n - h)

    if v > n - v:
        volume.append(v)
    else:
        volume.append(n - v)

    print(reduce(operator.mul, volume, 1))


if __name__ == '__main__':
    main()

File: problems/p/pieceofcake2/pieceofcake2.c
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <stdio.h>

int main(void) {
    int n, h, v, volume;

    if (scanf("%d %d %d\n", &n, &h, &v) != 3) {
        printf("Failed to read n, h and v!");
    }

    volume = 4;

    if (h > n - h) {
        volume *= h;
    }
    else {
        volume *= (n - h);
    }

    if (v > n - v)  {
        volume *= v;
    }
    else {
        volume *= (n - v);
    }

    printf("%d", volume);

    return 0;
}